### ISSUE TITLE
opam packaging

### ISSUE THREAD

**Issue by UnixJunkie** [arlencox/QUICr#2]
Hi,

If there was an install target in the Makefile, I could package this for opam.

**Comment by UnixJunkie** [comment]
I can also probably provide a different, simpler, build system.

**Comment by arlencox** [comment]
I do not believe that it is ready for release yet.  For this reason, I have not created an install target and have not created an opam package.  I know releasing it into opam doesn't have to imply that it is stable, but I do not want anyone using this expecting a polished library when it is definitely not.

I'm open to suggestions about the build system.  Here are the reasons why the build is the way it is today:
- I like using ocamlbuild/ocamlfind because it is well integrated into the OCaml ecosystem.  I do not wish to use omake, OCamlMakefile, or oasis.
- I like using packages to separate different parts of project into separate modules.  This enables replicating module names in different namespaces.  It also allows for potentially compiling them separately for dynamically loadable modules, which would be a benefit for the project (if it weren't such a pain managing plugins and things).
- I dislike using myocamlbuild.ml because it slows down the build in a measurable way.  However, if I could avoid having to specify .mlpack files because of a build using myocamlbuild, I would be amenable to that.

**Comment by UnixJunkie** [comment]
I was planning to do a "private" opam package (upon Xavier's request).
Which would be only visible to members of the memcad team at ENS.

**Comment by UnixJunkie** [comment]
I propose obuild: "the sane replacement to ocamlbuild". :)
The only valid reason to not use obuild is if you want to compile your thing under windows.
With obuild, builds are fast and the -j option is not a joke.

**Comment by arlencox** [comment]
Would it take long to do a trial of obuild?  I don't want to waste your time, but after 30 minutes of looking at it, I cannot figure out how to create the appropriate obuild.  I keep getting "helpful" errors like this:

```
uncaught exception
Fatal error: exception Prepare.Module.DependencyNoOutput
```

**Comment by UnixJunkie** [comment]
I will give it a shot, stay tuned. ;)

**Comment by UnixJunkie** [comment]
I could isolate the bug you found: https://github.com/ocaml-obuild/obuild/issues/134
Are you using ocamllex and ocamlyacc or menhir?

**Comment by arlencox** [comment]
I use ocamllex and ocamlyacc.

**Comment by arlencox** [comment]
I also added a license file (MIT).

**Comment by arlencox** [comment]
Have you made any progress on obuild?  I need to refactor the build a bit to optionally compile in some of the components.

**Comment by UnixJunkie** [comment]
no, the bug is still to be fixed, you can go on with your ocamlbuild setup for now

**Comment by arlencox** [comment]
Ok.  Thanks.
